src/Moq/ProxyFactories/CastleProxyFactory.cs

```diff
@@ -59,16 +59,21 @@ public CastleProxyFactory()
 		/// <inheritdoc />
 		public override object CreateProxy(Type mockType, Moq.IInterceptor interceptor, Type[] interfaces, object[] arguments)
 		{
+			// All generated proxies need to implement `IProxy`:
+			var additionalInterfaces = new Type[1 + interfaces.Length];
+			additionalInterfaces[0] = typeof(IProxy);
+			Array.Copy(interfaces, 0, additionalInterfaces, 1, interfaces.Length);
```

Comment:
It's a pity that this kind of thing happens twice now; once to add `IMocked<T>` for `Mock`, and once here, to add `IProxy` for `IInterceptor`. It would be nice to eventually optimize this a little.